pyrasterframes/src/main/python/pyrasterframes/rasterfunctions.py

```diff
@@ -781,6 +780,67 @@ def rf_identity(tile_col: Column_type) -> Column:
+
+def rf_focal_mean(tile_col: Column_type, neighborhood: Union[str, Column_type]) -> Column:
+    """Compute the mean value in its neighborhood of each cell"""
+    if isinstance(neighborhood, str):
+        neighborhood = lit(neighborhood)
+    return _apply_column_function('rf_focal_mean', tile_col, neighborhood)
+
+def rf_focal_median(tile_col: Column_type, neighborhood: Union[str, Column_type]) -> Column:
+    """Compute the max in its neighborhood value of each cell"""
+    if isinstance(neighborhood, str):
+        neighborhood = lit(neighborhood)
+    return _apply_column_function('rf_focal_median', tile_col, neighborhood)
+
+def rf_focal_min(tile_col: Column_type, neighborhood: Union[str, Column_type]) -> Column:
+    """Compute the min value in its neighborhood of each cell"""
+    if isinstance(neighborhood, str):
+        neighborhood = lit(neighborhood)
+    return _apply_column_function('rf_focal_min', tile_col, neighborhood)
+
+def rf_focal_mode(tile_col: Column_type, neighborhood: Union[str, Column_type]) -> Column:
+    """Compute the mode value in its neighborhood of each cell"""
+    if isinstance(neighborhood, str):
+        neighborhood = lit(neighborhood)
+    return _apply_column_function('rf_focal_mode', tile_col, neighborhood)
+
+def rf_focal_std_dev(tile_col: Column_type, neighborhood: Union[str, Column_type]) -> Column:
+    """Compute the standard deviation value in its neighborhood of each cell"""
+    if isinstance(neighborhood, str):
+        neighborhood = lit(neighborhood)
+    return _apply_column_function('rf_focal_std_dev', tile_col, neighborhood)
+
+def rf_moransI(tile_col: Column_type, neighborhood: Union[str, Column_type]) -> Column:
+    """Compute moransI in its neighborhood value of each cell"""
+    if isinstance(neighborhood, str):
+        neighborhood = lit(neighborhood)
+    return _apply_column_function('rf_focal_moransi', tile_col, neighborhood)
+
+def rf_aspect(tile_col: Column_type) -> Column:
+    """Calculates the aspect of each cell in an elevation raster"""
+    return _apply_column_function('rf_aspect', tile_col)
+
+def rf_slope(tile_col: Column_type, z_factor: Union[int, float, Column_type]) -> Column:
+    """Calculates slope of each cell in an elevation raster"""
+    if isinstance(z_factor, (int, float)):
+        z_factor = lit(z_factor)
+    return _apply_column_function('rf_slope', tile_col, z_factor)
+
+def rf_hillshade(tile_col: Column_type, azimuth: Union[int, float, Column_type], altitude: Union[int, float, Column_type], z_factor: Union[int, float, Column_type]) -> Column:
+    """Calculates the hillshade of each cell in an elevation raster"""
+    if isinstance(azimuth, (int, float)):
+        azimuth = lit(azimuth)
+    if isinstance(altitude, (int, float)):
+        altitude = lit(altitude)
+    if isinstance(z_factor, (int, float)):
+        z_factor = lit(z_factor)
+    return _apply_column_function('rf_hillshade', tile_col, azimuth, altitude, z_factor)
```

Comment:
Nicely done!!

Reply:
Sorry there's documentation duplicated. In another universe I would have figured out how to have the Python API generated from the annotations on the Expressions.